lib/spack/env/cc

```diff
@@ -187,6 +190,25 @@ if [[ -z $mode ]]; then
     done
 fi
 
+# This is needed to ensure we set RPATH instead of RUNPATH
+# (or the opposite, depending on the configuration in config.yaml)
+#
+# Documentation on this mechanism is lacking at best. A few sources
+# of information are (note that some of them take explicitly the
+# opposite stance that Spack does):
+#
+# http://blog.qt.io/blog/2011/10/28/rpath-and-runpath/
+# https://wiki.debian.org/RpathIssue
```

Comment:
This whole account is from the point of view of a single-prefix, system package manager.  Not one that has to maintain relationships between combinatorial sets of modules... so it's a completely different link situation.

Reply:
Agree completely. I put the links there more as reference for future developers to get some info on RPATH vs. RUNPATH rather than to explain Spack's philosophy towards linking. Let me know if you want me to remove them, or just rephrase the comment.